scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,546 @@
+package org.oppia.android.scripts.maven
+
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import org.oppia.android.scripts.maven.maveninstall.MavenListDependency
+import org.oppia.android.scripts.maven.maveninstall.MavenListDependencyTree
+import org.oppia.android.scripts.proto.License
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.OriginOfLicenses
+import org.oppia.android.scripts.proto.PrimaryLinkType
+import org.oppia.android.scripts.proto.SecondaryLinkType
+import java.io.File
+import java.io.FileInputStream
+import java.net.URL
+import java.util.concurrent.TimeUnit
+
+private const val WAIT_PROCESS_TIMEOUT_MS = 60_000L
+private const val LICENSES_TAG = "<licenses>"
+private const val LICENSES_CLOSE_TAG = "</licenses>"
+private const val LICENSE_TAG = "<license>"
+private const val NAME_TAG = "<name>"
+private const val URL_TAG = "<url>"
+
+/**
+ * Script to compile the list of the third-party Maven dependencies (direct and indirect both)
+ * on which Oppia-android depends.
+ *
+ * Usage:
+ *   bazel run //scripts:generate_maven_dependencies_list  -- <path_to_directory_root>
+ *   <path_to_maven_install_json> <path_to_maven_dependencies_textproto>
+ *
+ * Arguments:
+ * - path_to_directory_root: directory path to the root of the Oppia Android repository.
+ * - path_to_maven_install_json: absoulte path to the maven_install.json file.
+ * - path_to_maven_dependencies_textproto: absoulte path to the maven_dependencies.textproto
+ * that stores the list of maven dependencies compiled through the script.
+ * Example:
+ *   bazel run //scripts:generate_maven_dependencies_list -- $(pwd)
+ *   $(pwd)/third_party/maven_install.json $(pwd)/scripts/assets/maven_dependencies.textproto
+ */
+fun main(args: Array<String>) {
+  if (args.size < 3) {
+    throw Exception("Too less arguments passed.")
+  }
+  val pathToRoot = args[0]
+  val pathToMavenInstall = args[1]
+  val pathToMavenDependenciesTextProto = args[2]
+  runMavenRePinCommand(pathToRoot)
+  val bazelQueryDepsNames = runBazelQueryCommand(pathToRoot)
+  val finalMavenInstallList = readMavenInstall(pathToMavenInstall, bazelQueryDepsNames)
+
+  val dependenciesListFromTextproto = retrieveMavenDependencyList()
+  val dependenciesListFromPom =
+    getLicenseLinksFromPOM(finalMavenInstallList).mavenDependencyList
+
+  val licenseSetFromTextproto = mutableSetOf<License>()
+  val licenseSetFromPom = mutableSetOf<License>()
+
+  dependenciesListFromTextproto.forEach { dependency ->
+    dependency.licenseList.forEach {
+      licenseSetFromTextproto.add(it)
+    }
+  }
```

Comment:
More Kotlin idiomatic:

```suggestion
  val licenseSetFromPom = dependenciesListFromTextproto.flatMap { dependency ->
    dependency.licenseList
  }.toSet()
```

Ditto elsewhere (generally try to avoid vars/mutability where you can).

Reply:
Done.

Reply:
Verified.